### ISSUE TITLE
✅ 🔗 🚀 Feature Request: Add support for "YouTube Shorts" URLs in Elementor video components (Merged with #24220)

### ISSUE THREAD

**Issue by Nazrinn** [elementor/elementor#20330]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Video widget shows a blank block when the YouTube URL links to a short.

![2cf631afe110931c5b0fee5fda6eb397](https://user-images.githubusercontent.com/36900693/200628157-4c27a4e8-e851-4b2e-95f3-8474cd38f143.png)

This was tested on two different sites using Hello Elementor.

Nice-to-have: it would be nice if short videos immediately showed in portrait mode (otherwise the video looks broken).

### Steps to reproduce

1. Create a video widget
2. Add a short link, such as `https://www.youtube.com/shorts/smSlc6l4yHQ`, as the video widget YouTube URL.

Note that the issue can be circumvented by using the standard YouTube URL format, such as `https://www.youtube.com/watch?v=smSlc6l4yHQ`.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.3.34
	PHP Version: 8.0.18
	PHP Memory Limit: 32M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 512M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1
	Site URL: https://barbandcarole.ca
	Home URL: https://barbandcarole.ca
	WP Multisite: No
	Max Upload Size: 512 MB
	Memory limit: 512M
	Max Memory limit: 512M
	Permalink Structure: /blog/article/%postname%/
	Language: en-CA
	Timezone: America/Toronto
	Admin Email: h.desauln.pro@gmail.com
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.6.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_CA
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields PRO
		Version: 6.0.3
		Author: WP Engine

	Advanced Database Cleaner
		Version: 3.1.1
		Author: Younes JFR.

	Asset CleanUp: Page Speed Booster
		Version: 1.3.8.8
		Author: Gabe Livan

	Better Notifications for WP
		Version: 1.9.1
		Author: Made with Fuel

	Broken Link Checker
		Version: 1.11.19
		Author: WPMU DEV

	CB Change Mail Sender
		Version: 1.2.2
		Author: Md Abul Bashar

	Classic Editor
		Version: 1.6.2
		Author: WordPress Contributors

	Custom Post Type UI
		Version: 1.13.1
		Author: WebDevStudios

	Duplicator Pro
		Version: 4.5.8
		Author: Snap Creek

	Elementor
		Version: 3.8.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.8.1
		Author: Elementor.com

	Enable Media Replace
		Version: 4.0.0
		Author: ShortPixel

	Image optimization service by Optimole
		Version: 3.5.1
		Author: Optimole

	Lightweight Subscribe To Comments
		Version: 1.5.8
		Author: Isabel Castillo

	LiteSpeed Cache
		Version: 5.3
		Author: LiteSpeed Technologies

	Manage Notification E-mails
		Version: 1.8.4
		Author: Virgial Berveling

	Phoenix Media Rename
		Version: 3.8.8
		Author: crossi72

	Post Views Counter
		Version: 1.3.12
		Author: Digital Factory

	Simple Custom Post Order
		Version: 2.5.6
		Author: Colorlib

	Site Kit by Google
		Version: 1.87.0
		Author: Google

	SVG Support
		Version: 2.5.1
		Author: Benbodhi

	The SEO Framework
		Version: 4.2.6
		Author: The SEO Framework Team

	User Role Editor
		Version: 4.63.1
		Author: Vladimir Garagulya

	Wordfence Security
		Version: 7.7.1
		Author: Wordfence

	WordPress Importer
		Version: 0.8
		Author: wordpressdotorg

	WordPress Popular Posts
		Version: 6.0.5
		Author: Hector Cabrera

	WP-Sweep
		Version: 1.1.5
		Author: Lester 'GaMerZ' Chan

	WP Armour - Honeypot Anti Spam
		Version: 2.0.2
		Author: Dnesscarkey

	Yoast Duplicate Post
		Version: 4.5
		Author: Enrico Battocchi & Team Yoast


== Elements Usage ==
	
	archive : 3
		button : 5
		form : 1
		heading : 7
		html : 10
		image : 3
		shortcode : 6
		spacer : 1
		text-editor : 5
		theme-archive-title : 1
	footer : 1
		form : 1
		heading : 7
		image : 4
		shortcode : 4
		social-icons : 1
		text-editor : 6
	header : 1
		shortcode : 2
		theme-site-logo : 1
	section : 5
		button : 4
		heading : 4
		html : 6
		image : 2
		nav-menu : 1
		social-icons : 1
		text-editor : 4
		theme-site-logo : 1
	widget : 0
		html : 2
		image-carousel : 1
		nav-menu : 1
		shortcode : 2
		social-icons : 1
	wp-post : 46
		gallery : 2
		image : 128
		text-editor : 176
		video : 2
	wp-page : 7
		button : 30
		counter : 2
		divider : 2
		form : 2
		heading : 60
		html : 86
		image : 20
		media-carousel : 1
		shortcode : 18
		social-icons : 2
		spacer : 11
		text-editor : 42
	single-post : 2
		button : 2
		form : 1
		heading : 9
		html : 9
		post-comments : 1
		post-navigation : 1
		shortcode : 20
		text-editor : 1
		theme-post-content : 1
		theme-post-featured-image : 1
		video : 1
	error-404 : 1
		button : 1
		heading : 2


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Active
	Inline Font Icons: Inactive
	Accessibility Improvements: Active
	Additional Custom Breakpoints: Inactive by default
	Import Export Website Kit: Inactive
	Hide native WordPress widgets from search results: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Inactive by default
	Landing Pages: Inactive
	Color Sampler: Active by default
	Favorite Widgets: Active by default
	Admin Top Bar: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Inactive by default
	Form Submissions: Active
	Scroll Snap: Active by default


== Log ==
	
Log: showing 20 of 432022-10-03 01:37:56 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.7',
  'to' => '3.7.8',
)]
2022-10-03 17:31:01 [error] Save submission failed, db error: Table 'emrfro02_main.bcr_e_submissions' doesn't exist 
2022-10-31 01:49:44 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.8',
  'to' => '3.8.0',
)]
2022-10-31 01:49:45 [info] elementor::elementor_updater Started 
2022-10-31 01:49:45 [info] Elementor/Upgrades - _on_each_version Start  
2022-10-31 01:49:45 [info] Elementor/Upgrades - _on_each_version Finished 
2022-10-31 01:49:45 [info] Elementor/Upgrades - _v_3_8_0_fix_php8_image_custom_size Start  
2022-10-31 01:49:45 [info] Elementor/Upgrades - _v_3_8_0_fix_php8_image_custom_size Finished 
2022-10-31 01:49:45 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.8',
  'to' => '3.8.0',
)]
2022-10-31 01:50:19 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.7',
  'to' => '3.8.0',
)]
2022-10-31 01:50:19 [info] elementor-pro::elementor_pro_updater Started 
2022-10-31 01:50:19 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-10-31 01:50:19 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-10-31 01:50:19 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.7',
  'to' => '3.8.0',
)]
2022-11-07 21:58:38 [info] elementor-pro::elementor_pro_updater Started 
2022-11-07 21:58:38 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-11-07 21:58:38 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-11-07 21:58:38 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.8.0',
  'to' => '3.8.1',
)]
2022-11-07 21:58:38 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.8.0',
  'to' => '3.8.1',
)]
2022-11-07 21:58:39 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.8.0',
  'to' => '3.8.1',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Incompatible

== Elementor Pro - Compatibility Tag ==

				

````

**Comment by mcrowell** [comment]
I created a tutorial on how to embed YouTube Shorts inside Elementor:
https://www.youtube.com/watch?v=FFe8T93Xkh0

**Comment by nicholaszein** [comment]
Hello @Nazrinn!

Thanks for submitting your issue! 🙏 

This is more of a feature request than a bug since Elementor did not implement support for YouTube Shorts, therefore there is nothing broken. Although as you and @mcrowell mentioned, there is a workaround to make this work by removing the string `shorts/` from the URL.

As for the **aspect ratio**, you can define it in the Widget Styles. There are no plans to automate this for the time being.

I've changed the title of this issue to reflect what I said above.

Please understand that while we do welcome new feature requests, we can't promise they will be considered or added to our roadmap for the time being.

We will, of course, keep this suggestion in mind and consider adding it if there is significant demand in the future.  

Please feel free to contact us again with other feature suggestions. 😉 

Best regards

**Comment by Nazrinn** [comment]
Hi @nicholaszein!

Yes I filled it under "bug" because that's how my clients perceived it. I'm aware there is a workaround, but for people like my not-too-tech-savvy who like to have the liberty to make their own blog posts, it looked like a bug.

Just out of curiosity, would it be possible to send some kind of pull request in the future if I feel inclined enough to fix the issue myself at the code level?

**Comment by nicholaszein** [comment]
Hello @Nazrinn!

### That would be AWESOME! 💪

We love PRs from our users! If you're up to the task we would love to receive your PR and eventually implement it.

You can learn how to contribute by following the [Contribution Guidelines](https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md).

I hope to see your contribution soon! 🙂 

Best regards

**Comment by nicholaszein** [comment]
Hi there,

### 📅  This request is scheduled to be Migrated to the new [Discussions](https://github.com/elementor/elementor/discussions/categories/feature-request) section.

Please do not duplicate this request.

Once we migrate it, we encourage you to cast your **⬆️ Upvote** on the first comment if you feel it should be considered for our roadmap.

**Comment by nicholaszein** [comment]
Hi there,

### 🔃 This Feature Request is being merged with #24220, which is on the same subject. 

if you still think this request should be considered for our roadmap, please cast your **⬆️ Upvote** on the first comment.

Don't worry, your comments, reactions, and participation in this issue will be accounted for when considering this Feature Request for our Roadmap.

Happy voting! 🙂

**Comment by nicholaszein** [comment]
Hi there!

We have great news! 🙌 

### 📢 We're happy to announce that YouTube Shorts support for the Video widget was released with Elementor Core v3.27.0! 🥳 

You can now share YouTube shorts in the **Video widget**, and even change the aspect ratio. Just select the **`Source`** as **`YouTube`**, and paste your YouTube shorts link. Then in the **`Style`** tab you can change the Aspect Ration to 9:16.

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
